buildlib/travis-build

```diff
@@ -26,8 +22,23 @@ ninja
 
 # Test with coherent DMA mode disabled (ie as would be on ARM32, etc)
 cd ../build-clang
+cp ../util/udma_barrier.h ../util/udma_barrier.h.old
 echo "#error Fail" >> ../util/udma_barrier.h
 rm CMakeCache.txt
 CC=clang-3.9 CFLAGS=-Werror cmake -GNinja ..
 ninja
+
+# Finally run through gcc-6 64 bit through the debian packaging This gives a
+# good clue if patches are changing packaging related things, the RPM stuff
+# will have to be audited by hand.
+
+# When running cmake through debian/rules it is hard to set -Werror,
+# instead force it on by changing the CMakeLists.txt
 cd ..
+cp util/udma_barrier.h.old util/udma_barrier.h
+echo 'set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror")' >> buildlib/RDMA_EnableCStd.cmake
+sed -i -e 's/-DCMAKE_BUILD_TYPE=Release//g' debian/rules
+sed -i -e 's/ninja \(.*\)-v/ninja \1/g' debian/rules
+
+CC=gcc-6 debian/rules build
```

Comment:
Hm, I ran it that way because I thought dpkg-buildpackage scrubbed the environment and would eat my CC=, but that is not true.

The downside is that dpkg-buildpackage does some unnecessary work for the travis environment, do you think it is worth changing?